cmd/src/server.rs

```diff
@@ -79,11 +72,29 @@ pub fn run_tikv(mut config: TiKvConfig) {
         "cluster_id" => cluster_id
     );
 
+    let (_, config) = ConfigClient::create(pd_client.clone(), &config)
+        .unwrap_or_else(|e| fatal!("failed to register config to pd: {}", e));
+
+    // Print version information.
+    tikv::log_tikv_info();
+    info!(
+        "using config";
+        "config" => serde_json::to_string(&config).unwrap(),
+    );
+
+    config.write_into_metrics();
+    // Do some prepare works before start.
+    pre_start(&config);
+
     let _m = Monitor::default();
     run_raft_server(pd_client, &config, security_mgr);
 }
 
-fn run_raft_server(pd_client: RpcClient, cfg: &TiKvConfig, security_mgr: Arc<SecurityManager>) {
+fn run_raft_server(
+    pd_client: Arc<RpcClient>,
+    cfg: &TiKvConfig,
+    security_mgr: Arc<SecurityManager>,
+) {
```

Comment:
This PR is conflict with #6008 . Cc @nrc

Reply:
I think it will be fairly mechanical to rebase.